### ISSUE TITLE
Cannot read property 'ImageClass' of undefined

### ISSUE THREAD

**Issue by ChucKN0risK** [canvg/canvg#627]
### The issue
I'm building an app with [electron-vue](https://github.com/SimulatedGREG/electron-vue). While I was calling it this error appeared in my console.

```
/Applications/MAMP/htdocs/Perso/specify-app/node_modules/canvg/dist/node/canvg.js:88 Uncaught TypeError: Cannot read property 'ImageClass' of undefined
    at canvg (/Applications/MAMP/htdocs/Perso/specify-app/node_modules/canvg/dist/node/canvg.js:88)
```

### The code
MySingleFileComponent.vue:

```
<template>
    <canvas
      class="u-hide"
      id="canvas"
      width="400px"
      height="400px">
    </canvas>
</template>

<script>
  export default {
    methods: {
      exportIcon() {
        canvg(this.$el.querySelector('#canvas'), svg);
      },
  };
</script>
```
### The solution

Adding an empty `Object` as a third parameter:
`canvg(this.$el.querySelector('#canvas'), svg, {});`

**Comment by gabelerner** [comment]
makes sense, although i don't know if images would work for you with an empty object and i need to fix this in the build. for now you can just call `canvg` with
`{ ImageClass: Canvas.Image }` where `Canvas = require("canvas")`.

**Comment by webarchymeta** [comment]
@gabelerner , the fix may not be general enough since in electronjs environment, node and the browser is mixed together, there is no need for a dependency on node-canvas, which contains native module that would reduce the portability of the resulting electronjs application and comlicates the build process.

IMHO, it's better to provide certain callback for the user to provide the Canvas and Image object, or using other dependency injection mechanism ...

**Comment by gabelerner** [comment]
what i do know is that node works separately and browser works separately, is there something weird about electron?

We can do feature sniffing instead of using rollup to build per your suggestion but it doesn't work in all cases since a server environment might have `document.createElement` defined someway.  that's why we just prebuild two separate versions.

**Comment by webarchymeta** [comment]
Ah, that's what I don't know (the predefined `document.createElement` part). I am developing client side applications using electron ... Maybe a third version for mixed environment, like the electronjs, in the future?

**Comment by gabelerner** [comment]
do you only need the client side version? that's defined in https://github.com/canvg/canvg/blob/master/package.json#L19 under the browser section.  it seems like it's not yet supported in electron so maybe channel your questions towards https://github.com/electron/electron/issues/1574

**Comment by webarchymeta** [comment]
I tried that the "browser" version, it somehow does not work as the demo. I did not dig any further as to why. In our application, we need to read the svg data from local disk using node `fs` then draw it into a canvas (in background) and then stream it to remote nodes via webrtc. The server side version of your lib seems to work better for us. My current temporary solution to the "problem" of our is "customization": make a copy of your file, applied the above change and include it into the project.

I made further customization with respect to the svg size initialization procedure according to our layout requirement: namely, we would like to have the parent of the canvas to decide, dynamically during overall layout, the size of the svg image and resize (or fit) the svg image via the viewport accordingly. The current version of the lib seems to be not working that way ... But, I am not familiar with the internal workings of svg, so I could be misunderstanding ...
